test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py

```diff
@@ -93,28 +100,26 @@ def _parse_python_script_to_test_case_declarations(
 
 
 def __parse_python_tests(
-    python_test_files: list[Path], python_test_version: str, mandatory: bool
+    python_test_version: str, mandatory: bool
 ) -> list[PythonSuiteDeclaration]:
     suites = _init_test_suites(python_test_version)
 
-    for python_test_file in python_test_files:
-        test_cases = _parse_python_script_to_test_case_declarations(
-            python_test_path=python_test_file,
-            python_test_version=python_test_version,
-        )
+    test_cases = _parse_python_script_to_test_case_declarations(
+        python_test_version=python_test_version,
+    )
 
-        for test_case in test_cases:
-            python_test_type = test_case.class_ref.python_test.python_test_type
-            if mandatory:
-                if python_test_type == PythonTestType.MANDATORY:
-                    suites[SuiteType.MANDATORY].add_test_case(test_case)
-            else:
-                if python_test_type == PythonTestType.COMMISSIONING:
-                    suites[SuiteType.COMMISSIONING].add_test_case(test_case)
-                elif python_test_type == PythonTestType.NO_COMMISSIONING:
-                    suites[SuiteType.NO_COMMISSIONING].add_test_case(test_case)
-                elif python_test_type != PythonTestType.MANDATORY:
-                    suites[SuiteType.LEGACY].add_test_case(test_case)
+    for test_case in test_cases:
+        python_test_type = test_case.class_ref.python_test.python_test_type
+        if mandatory:
```

Comment:
This if statement seems redundant. The test case already has the information in `python_test_type`. Maybe this parameter is not needed in this method

Reply:
It's not... But we need to improve the process to perform only one pass over the tests and separate the collections correctly.

Currently, the tests are parsed twice to do this (mandatory collection and Python collection).

We can improve it, yes. But I think it's better to do this in another PR.